### ISSUE TITLE
Should the index lifecycle plugin name be changed to "ilm"

### ISSUE THREAD

**Issue by colings86** [elastic/elasticsearch#33265]
We changed all the ILM endpoints from using `index_lifecycle` to `ilm` but the name of the ILM plugin is still `index_lifecycle`. We should maybe change this to `ilm` as well?

/cc @jasontedor @talevy @dakrone

**Comment by jasontedor** [comment]
Yes, I thought we discussed this and I only did not pick it up with the initial index_lifecycle -> ilm change because

**Comment by colings86** [comment]
ok, I'll open a PR for this shortly

**Comment by jasontedor** [comment]
Would you clarify what you mean by the name? Looking now at the commit I pushed, I think I addressed what I think is the name of the plugin to `ilm`.

**Comment by colings86** [comment]
Sure, I mean:

https://github.com/elastic/elasticsearch/blob/index-lifecycle/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackField.java#L35

This is used in the `GET _xpack` API as the name of the feature. 

Much lower priority but we should probably also change:

https://github.com/elastic/elasticsearch/blob/index-lifecycle/x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/indexlifecycle/IndexLifecycle.java#L93

This is use in the scheduler job name, though its probably best to change it to just use `XPackField.INDEX_LIFECYCLE` and remove `IndexLifecycle.NAME` completely.

**Comment by jasontedor** [comment]
Yes, we should definitely change those. By the name, I was referring to the name of the project (`ilm`) and the archive name (`x-pack-ilm`) and what appears in the startup logs:

```
[2018-08-30T07:18:41,561][INFO ][o.e.p.PluginsService     ] [] loaded module [x-pack-ilm]
```

**Comment by colings86** [comment]
I opened https://github.com/elastic/elasticsearch/pull/33270#issuecomment-417293410 to address this

**Comment by colings86** [comment]
fixed in #33270
